### ISSUE TITLE
Implement simple colour inline image fall backs while loading

### ISSUE THREAD

**Issue by brendanheywood** [gthomas2/moodle-filter_imageopt#32]
When we process the image and resize it we can also do further processing and reduce the image to a single colour (or maybe a set of 4 colors) and then we render these inline as part of the placeholder either as a background color or an inline data url which then gets swapped out when it lazy loads. More useful when the threshold for loading is within the visible bounds, or when you are scrolling fast and the loading can't keep up. You would still have some sort of lazy loading spinner probably the bootstrap one as well.

**Comment by gthomas2** [comment]
Hi Brendan. I love the single color idea. Maybe we could overlay the existing image placeholder over a background of the single color.

**Comment by brendanheywood** [comment]
Yup the loading spinner on top of the fixed color or gradient will be nice
